### ISSUE TITLE
New Theme - Tracks size scaling

### ISSUE THREAD

**Issue by WinLin97** [LMMS/lmms#2899]
Hi all,

New skin is very, very cool, but...

The idea is to allow setting the default size of tracks and that did not have to diminish and zoom all. Default size of tracks is for me too big. In my opinion this option is probably to implement in style.css :)
![przechwycenie obrazu ekranu_2016-07-07_15-38-09](https://cloud.githubusercontent.com/assets/10725180/16658040/18192df6-4465-11e6-8990-46be4990de70.png)

Best regards :)

**Comment by WinLin97** [comment]
OK, I created a patch, for code :) To apply it type in terminal: 
patch /lmms-source-code/include/Track.h -i lmms.patch

https://drive.google.com/file/d/0B-q0QyhmQ3_FNFl2ejIwclZCaXM/view?usp=sharing

**Comment by tresf** [comment]
@WinLin97, thanks.

We encourage pull-requests for these things now (well actually... this commit is so small, you could do it through the web browser).

https://github.com/LMMS/lmms/wiki/Submitting-a-patch

But can you provide more information on this patch?  How does making the minimal height smaller fix the aforementioned issue?

```diff
--- Track.h	2016-07-08 14:10:28.117878811 +0200
+++ Track.h	2016-07-08 14:10:14.196174924 +0200
@@ -67,8 +67,8 @@
  * Tracks can be resized by shift-dragging anywhere inside the track
  * display.  This sets the minimum size in pixels for a track.
  */
-const int MINIMAL_TRACK_HEIGHT = 32;
-const int DEFAULT_TRACK_HEIGHT = 32;
+const int MINIMAL_TRACK_HEIGHT = 5;
+const int DEFAULT_TRACK_HEIGHT = 21;
 
 const int TCO_BORDER_WIDTH = 2;
 ```

**Comment by tresf** [comment]
Ok, I realize what you're requesting now.  @WinLin97 don't you think we could make this feature a bit more... cosmetic?

**Comment by WinLin97** [comment]
I think the most :) I am currently at the stage of getting to know the structure of the program :) And do not quite know how to do this function in the LMMS Setup or .lmmsrc.xml or style.css. I did it as much as I could.

**Comment by tresf** [comment]
@WinLin97 is this a regression?  e.g. Did 1.1.3 allow the resizing of tracks or is this just offering the ability through CSS by removing a hard-coded value?

**Comment by WinLin97** [comment]
So, why default size of tracks is set "hard"? And I could change it only in code?

**Comment by tresf** [comment]
> In my opinion this option is probably to implement in `style.css` or in LMMS configuration file/window :)

I assume you're using CSS then?  How does the patch work?

**Comment by WinLin97** [comment]
No. Simply I assume that you can add this option to the configuration file of style (style.css).

**Comment by tresf** [comment]
@WinLin97 I'm not sure I understand.

Please issue a new Pull Request with the mentioned feature with explanation of how to use it and we'd be happy to review and provide feedback.

**Comment by WinLin97** [comment]
Super, so You will have to wait long, because I do not know how to implement it...
But...
I want It would look something like this :)
This allow to change the default height, without digging in the code.

**Comment by Umcaruje** [comment]
The problem is that the actual widgets in the track don't respond at all on the track height change, and simply changing the default size won't cut it and can cause issues. I'll look into this.

**Comment by WinLin97** [comment]
I personally in beta version (and of course in stable-1.1.3) dry a thing does not bother, I use it that way :)
![lmms](https://cloud.githubusercontent.com/assets/10725180/16691716/0d26cc54-452e-11e6-9d3e-04eaac321a0d.png)
More tracks is located at display :)

**Comment by BaraMGB** [comment]
Yes, but the widgets in the track control are cropped, especially the knob label are invisible.

**Comment by WinLin97** [comment]
In this case, to be visible (VOL, PAN), they would have to be placed on the sides, on the bottom there is no place for them...

**Comment by tresf** [comment]
>  especially the knob label are invisible.

That's inevitable though.

> Yes, but the widgets in the track control are cropped,

True although the flat theme makes that much less noticeable.  @WinLin97 what can we do to help with the coding? :+1:

**Comment by WinLin97** [comment]
Tell me in whose files I can add these options in the setup :) (Which files I need to edit, to do this?)
![lmms](https://cloud.githubusercontent.com/assets/10725180/16692619/e9035618-4531-11e6-8e95-88767e2e3ad1.png)

**Comment by DeRobyJ** [comment]
I believe the setup file is here, if that's what you were searching: https://github.com/LMMS/lmms/blob/master/src/gui/SetupDialog.cpp

**Comment by tresf** [comment]
`include/SetupDialog.h ` as well as `src/gui/SetupDialog.cpp`, specifically somewhere around here: [`SetupDialog.cpp#L342`](https://github.com/LMMS/lmms/blob/master/src/gui/SetupDialog.cpp#L342)

**Comment by WinLin97** [comment]
I can not do that. For me this is too hard... Errors, errors, errors, when I compile it.

**Comment by ImGGAAVVIINN** [comment]
any updates?
